pkg/util/cert/certutil.go

```diff
@@ -132,9 +132,7 @@ func DecodeCertificate(raw []byte) (cert *x509.Certificate, err error) {
 	if err != nil {
 		return nil, err
 	}
-	if len(certs) != 1 {
-		return nil, errors.New("only one certificate should be present, more found")
-	}
```

Comment:
Can you explain a bit why we should remove this check and always return the first item in the list?

Reply:
Thank you for the review, yes.
This ParseCertificates function parse the intermediate and leaf certificates in an array: https://github.com/banzaicloud/koperator/blob/1bf8374c16ba6627606597f333260c43641a32f6/pkg/util/cert/certutil.go#L68
Certs should be in this order:  leaf -> intermediate(s) -> root. The first is the leaf certificate that we need when we want to get the Distinguish Name. When multiple certificates are found it should not be handled as an error.
Koperator's built in kafka client can handle when there are intermediate certificate in the tls.crt. The kcat also can handle. https://github.com/banzaicloud/koperator/blob/1bf8374c16ba6627606597f333260c43641a32f6#L294 and https://github.com/banzaicloud/koperator/blob/1bf8374c16ba6627606597f333260c43641a32f6/pkg/util/util.go#L440

Reply:
This will be nil: https://github.com/banzaicloud/koperator/blob/1bf8374c16ba6627606597f333260c43641a32f6/pkg/util/pki/common.go#L99 so the return cert.Subject.String() will be nil pointer dereference.

Reply:
Thanks for explaining!